src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -0,0 +1,66 @@
+import { createSelector, createSlice } from '@reduxjs/toolkit'
+import { CAIP19 } from '@shapeshiftoss/caip'
+import isEqual from 'lodash/isEqual'
+import { ReduxState } from 'state/reducer'
+
+// TODO(0xdef1cafe): implement this in @shapeshiftoss/caip first
+// caip10 is account spec
+// https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md
+export type CAIP10 = string
+
+export type PortfolioAccounts = {
+  byId: {
+    // asset ids belonging to an account
+    [k: CAIP10]: CAIP19[]
+  }
+  // a list of accounts in this portfolio
+  ids: CAIP10[]
+}
+
+export type PortfolioBalances = {
+  byId: {
+    // balance in base units of asset - bn doesn't serialize
+    [k: CAIP19]: string
```

Comment:
@natven excellent point about multiple eth accounts, come up with a way to manage the same asset held across multiple accounts, as this will flatten balances